python/lsst/faro/base/CatalogMeasurementBase.py

```diff
@@ -16,15 +61,83 @@ class CatalogMeasurementBaseTaskConfig(MetricConfig,
+                CatalogMeasurementBaseConfig.referenceCatalogLoader, self, msg)
+
 
 class CatalogMeasurementBaseTask(MetricTask):
+    """Base class for science performance metrics measured from source/object catalogs."""
 
-    ConfigClass = CatalogMeasurementBaseTaskConfig
+    ConfigClass = CatalogMeasurementBaseConfig
     _DefaultName = "catalogMeasurementBaseTask"
 
     def __init__(self, config, *args, **kwargs):
         super().__init__(*args, config=config, **kwargs)
         self.makeSubtask('measure')
 
-    def run(self, cat):
-        return self.measure.run(cat, self.config.connections.metric)
+    def run(self, catalog, **kwargs):
+        return self.measure.run(self.config.connections.metric, catalog, **kwargs)
+
+    def _getReferenceCatalog(self, butlerQC, dataIds, refCats, filterList, epoch=None):
+        """Load reference catalog in sky region of interest.
+
+        Parameters
+        ----------
+        butlerQC : `lsst.pipe.base.butlerQuantumContext.ButlerQuantumContext`
+            Butler quantum context for a Gen3 repository.
+        dataIds: interable of `lsst.daf.butler.dataId`
+             An iterable object of dataIds that point to reference catalogs
+             in a Gen3 repository.
+        refCats : iterable of `lsst.daf.butler.DeferredDatasetHandle`
+            An iterable object of dataset refs for reference catalogs in
+            a Gen3 repository.
+        filterList : `list` [`str`]
+            List of camera physicalFilter names to apply color terms.
+        epoch : `astropy.time.Time`, optional
+            Epoch to which to correct proper motion and parallax
+            (if available), or `None` to not apply such corrections.
+
+        Returns
+        -------
+        refCat : `lsst.afw.table.SimpleCatalog`
+            Catalog of reference objects from region.
+        refCatCorrected : `numpy.ndarray`
+            Catalog of reference objects with proper motions and color terms
+            (optionally) applied.
+        """
+        center = lsst.geom.SpherePoint(butlerQC.quantum.dataId.region.getBoundingCircle().getCenter())
+        radius = butlerQC.quantum.dataId.region.getBoundingCircle().getOpeningAngle()
+
+        loaderTask = LoadReferenceCatalogTask(
+            config=self.config.referenceCatalogLoader,
+            dataIds=dataIds,
+            refCats=refCats)
+
+        # Get catalog with proper motion and color terms applied
+        refCatCorrected = loaderTask.getSkyCircleCatalog(center, radius,
+                                                         filterList,
+                                                         epoch=epoch)
+
+        # Get unformatted catalog w/ all columns
+        skyCircle = loaderTask.refObjLoader.loadSkyCircle(center, radius,
+                                                          loaderTask._referenceFilter,
+                                                          epoch=epoch)
+        if not skyCircle.refCat.isContiguous():
```

Comment:
This check should not be necessary any more: loadSkyCircle should do this for you.

Reply:
Thanks. This does seem to be the case looking at `loadSkyCircle`. Removed.